### PR TITLE
Typescript version

### PR DESCRIPTION
**Pull request by jenni-divvito**
Hey there @viRingbells, thanks very much for your work on this lib, it was exactly what I needed for my current task. My only problem was that because of the use of native async functions I could not use the library in the environment I am using (Google Firebase cloud) as it is still on Node 6.

I made this short fork to put it into typescript, so that the native async functions are transpiled to work in older versions of Node. Not sure if you will want to merge it or not but thought I would give you the option.

I ensured all the tests still pass without modification after changing to typescript, and the API should be unchanged for outside users.

A note that I changed the package name only so that I could use the fork without conflicts in the meantime.

Have a good day :)

### CODE REVIEW

**Comment by viRingbells** [comment]
Merged. That looks great, thanks for your pull request!

**Comment by jenni-divvito** [comment]
Awesome, thanks very much!

Note that this merge has changed the package name and author in package.json (sorry, I had those different on my fork so that my package manager would not use the cached version in the meantime until it is up on NPM) - you may need to change that back for your version here.

File: package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-  "name": "async-stream-reader",
-  "version": "1.0.0",
+  "name": "async-stream-reader-divvito",
```
Comment (jenni-divvito) [inline package.json]: This should probably change back to "async-stream-reader"

File: package.json [hunk 2]
```diff
@@ -23,11 +23,12 @@
...
     "pretest": "npm run test:clear-env && npm run lint && npm run test:prepare-env",
     "test": "./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha -- --require should --recursive test",
-    "posttest": "npm run test:clear-env"
+    "posttest": "npm run test:clear-env",
+    "build": "tsc"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/viRingbells/async-stream-reader.git"
+    "url": "git+https://github.com/divvito/async-stream-reader.git"
```
Comment (jenni-divvito) [inline package.json]: This too

File: package.json [hunk 3]
```diff
@@ -39,7 +40,7 @@
   "author": "Sun Haohao",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/viRingbells/async-stream-reader/issues"
+    "url": "https://github.com/divvito/async-stream-reader/issues"
```
Comment (jenni-divvito) [inline package.json]: And these two

**Comment by viRingbells** [comment]
OK. Updated in version 1.2.0 on npm.

**Pull request merged**